payload/context/register_test.go

```diff
@@ -19,53 +19,62 @@ import (
 
 type registerSuite struct {
 	testing.IsolationSuite
+
+	hookCtx *stubRegisterContext
+	command RegisterCmd
 }
 
 var _ = gc.Suite(&registerSuite{})
 
-func (registerSuite) TestInitNilArgs(c *gc.C) {
-	r := RegisterCmd{}
-	err := r.Init(nil)
+func (s *registerSuite) SetUpTest(c *gc.C) {
+	s.IsolationSuite.SetUpTest(c)
+
+	s.hookCtx = &stubRegisterContext{}
+
+	s.command = RegisterCmd{
+		hookContextFunc: func() (Component, error) {
+			return s.hookCtx, nil
+		},
+	}
+}
+
+func (s registerSuite) TestInitNilArgs(c *gc.C) {
+	err := s.command.Init(nil)
```

Comment:
I'll change the receivers but no, all the changes here are related to how the command is constructed for the tests and how the tests get it.